src/native/corehost/corehost.cpp

```diff
@@ -312,6 +312,12 @@ int __cdecl wmain(const int argc, const pal::char_t* argv[])
 int main(const int argc, const pal::char_t* argv[])
 #endif
 {
+// Disable Swift backtrace on macOS to avoid conflicts with the host's signal handlers.
+// see https://github.com/dotnet/runtime/issues/118823
+#if defined(TARGET_OSX)
+    ::setenv("SWIFT_BACKTRACE", "enable=no", 1);
```

Comment:
Should we set this only if `SWIFT_BACKTRACE` is not `empty()`? That way if someone really wants it enabled, they can do `SWIFT_BACKTRACE="enable=yes" dotnet run` to turn it back on again.

Reply:
Or easier, set the `overwrite` parameter to `0`.